spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala

```diff
@@ -93,6 +99,22 @@ object DataFrameImplicits {
      * @return Returns aligned and filtered utils
      */
     def alignSchema(structType: StructType): DataFrame = alignSchema(structType.getDataFrameSelector())
+
+    /**
+     * Persist this Dataset with the default storage level (`MEMORY_AND_DISK`), the same way as
+     * [[org.apache.spark.sql.Dataset.cache]] does. But does not throw a warning if the DataFrame has been cached before.
+     *
+     * @return the DataFrame itself
+     * @since 0.3.0
+     */
+    def cacheIfNot(): DataFrame = {
```

Comment:
Do you think we could unit test this? 

Something like
- assert not cached
- run cacheIfNot 
- assert cached
- run cacheIfNot
- asser nothing bad happend

Reply:
I am not convinced it would test what the function offers, not until we have some log interceptor (for which I really long for).
Also not sure how to test "_assert nothing bad happend_, furthermore" based on w white-box testing it's kind checking a piece of code is skipped 🤷

Reply:
Use the same logic for finding if it is already cached as you use in the method. Seems duplicitous or shifty for your case here. But if somebody starts playing with the main method and it is not exactly it, the unit test will stop him.